### DISCUSSION TITLE
Why is the UI so different on Android and iOS? Am I doing something wrong?

### DISCUSSION THREAD

**Discussion by pookyxpook** [dotnet/maui#26480] [Q&A]
Hello everyone,
I've been trying to develop an app on Android and iOS. I developed by using an android emulator the whole time until I kinda finished the first phase. I just tried my app on iOS simulator not long ago and I found out that the layout on each page is so messed up on iOS. I did expect some differences but I didn't expect to see the layout this messed up on iOS. You can see in the pics. Can anyone suggest what I should do? Create a new page for each page for iOS? Did I violate some guideline so I ended up with this messed up layout on iOS?
Thank you in advance. Any suggestions would be very appreciated.

![1](https://github.com/user-attachments/assets/38d045f2-b69e-4e78-a354-2f5c038c9920)

![2](https://github.com/user-attachments/assets/00b9dff9-88a2-4e19-8c85-2d7c165fd664)

![3](https://github.com/user-attachments/assets/25aceaa0-23ee-4d11-884d-55be550b5f94)

![4](https://github.com/user-attachments/assets/0c90dca9-a910-4549-b95c-6fbec9aa78ba)

![5](https://github.com/user-attachments/assets/3a4f8923-8b41-45f9-bb45-7bc19cb3f48c)

**Comment by ljzj2** [comment]
it is feature not a bug

**Comment by AlleSchonWeg** [comment]
This is normal. I have the same problems.
Every platform you will receive different layouts. It's because of bugs and different behaviours of the native controls. 
I have no simple solution to this. Just test Android and iOS parallel. Flutter perhaps is better, because the UI is with drawn with skia. This has also problems....

**Reply by pookyxpook** [reply]
ah ok at least it's good to know that it's normal so I can deal with it. I really appreciate your answer. Do you happen to know if there's any guideline or best practice for this? Using OnPlatform in xaml? Or is there a better cleaner way to do it?

**Reply by AlleSchonWeg** [reply]
As far as i know there is no best practice. It depends how complex your layout is. Very simple hello world pages could look the same on android/ios.
In my app i have a collectionview which works on android with ItemSizingStrategy=MeasureAllItems but on iOS i need to set it to MeasureFirstItem to make it work.
I learned this:
If i make a change in the UI Code i test it on android and ios. The main mistake -in my opinion- is to develop an app only with android and if finished starting with iOS.

**Reply by pookyxpook** [reply]
> As far as i know there is no best practice. It depends how complex your layout is. Very simple hello world pages could look the same on android/ios. In my app i have a collectionview which works on android with ItemSizingStrategy=MeasureAllItems but on iOS i need to set it to MeasureFirstItem to make it work. I learned this: If i make a change in the UI Code i test it on android and ios. The main mistake -in my opinion- is to develop an app only with android and if finished starting with iOS.

Sorry for my late response and thank you so much again for your input. I totally agreed. I should have started testing on iOS from the very beginning.

**Comment by eqmarcin** [comment]
Try to use grid as much as possible. Usually I have UI issues with vertical and horizontal stacks in page layout so grid works better for me on both platforms
